python/cuml/cuml/metrics/_classification.py

```diff
@@ -13,16 +13,121 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import warnings
 
-from cuml.internals.input_utils import input_to_cupy_array
 import cuml.internals
+from cuml.internals.input_utils import input_to_cupy_array
 from cuml.internals.safe_imports import cpu_only_import
 from cuml.internals.safe_imports import gpu_only_import
 
 cp = gpu_only_import("cupy")
+cudf = gpu_only_import("cudf")
 np = cpu_only_import("numpy")
 
 
+def _input_to_cupy_or_cudf_series(x, check_rows=None):
+    """Coerce the input to a 1D cupy array or cudf Series.
+
+    For classification problems we need to support the full range
+    of supported input dtypes. cupy cannot support string labels,
+    and cudf cannot support float16. To handle this, we prefer cudf
+    if the input is cudf, otherwise try to coerce to cupy, falling
+    back to cudf if the dtype isn't supported.
+    """
+    if isinstance(x, cudf.Series):
+        # Drop the index so comparisons don't try to align on index
+        out = x.reset_index(drop=True)
+        n_cols = 1
+    else:
+        try:
+            out, _, n_cols, _ = input_to_cupy_array(x)
+            out = out.squeeze()  # ensure 1D
+        except ValueError:
+            # Unsupported dtype, use cudf instead
+            # Drop the index so comparisons don't try to align on index
+            out = cudf.Series(x, nan_as_null=False, copy=False).reset_index(
+                drop=True
+            )
+            n_cols = 1
+
+    n_rows = len(out)
+
+    if n_cols > 1:
+        raise ValueError(f"Expected 1 columns but got {n_cols} columns.")
+    if check_rows is not None and n_rows != check_rows:
+        raise ValueError(f"Expected {check_rows} rows but got {n_rows} rows.")
+
+    return out
+
+
+@cuml.internals.api_return_any()
+def accuracy_score(
+    y_true, y_pred, *, sample_weight=None, normalize=True, **kwargs
+):
+    """
+    Accuracy classification score.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        Ground truth (correct) labels.
+    y_pred : array-like of shape (n_samples,)
+        Predicted labels.
+    sample_weight : array-like of shape (n_samples,)
+        Sample weights.
+    normalize : bool
+        If ``False``, return the number of correctly classified samples.
+        Otherwise, return the fraction of correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        The fraction of correctly classified samples, or the number of correctly
+        classified samples if ``normalize == False``.
+    """
+
+    if kwargs:
+        warnings.warn(
+            "`convert_dtype` and `handle` were deprecated from `accuracy_score` "
+            "and will be removed in a future release.",
+            FutureWarning,
+        )
```

Comment:
We added a similar warning in the `r2_score` PR, doing the same here.

Reply:
Can we agree a release in which they will go away? I have a preference for warnings that say "foo was changed in version X and will be removed in Y" - more concrete and less "left as exercise for the reader"

Reply:
No strong thoughts except:
- I don't know when we'd want to drop it
- `cuml` currently has several deprecation warnings that don't list a removal version (including the equivalent one I added for `r2_score`).

I think adding a version it will be removed is nice _if_ there's an obvious policy that makes determining that easy for developers. If there's not, "perfect is the enemy of good" IMO.

Reply:
> in this case I support "changed in 25.04 and removed in 25.06."

Works for me.

It also works for me as a general policy, so that we don't have to have this discussion every time we deprecate something.

Reply:
Happy to adopt that. This also means that we should _actually_ drop things in those releases. I'll try to track that.

Reply:
A reason to add the version to the warning message, then you can grep for all deprecations for the upcoming release.